sympy/core/relational.py

```diff
@@ -1514,3 +1562,103 @@ def split_real_imag(expr):
             rv = False
         if rv is not None:
             return rv
+
+
+def _simplify_inequality_with_minmax(e, fn, cmp1, cmp2):
+    """ Simplify inequalities, i.e., ``>``, ``<``, ``>=``, ``<=``, that
+    contains a :class:`~.Min` and/or :class:`~.Max` functions.
+
+    Written so that only ``>`` and ``>=``are supported, but it is possible to
+    ``<`` and ``<=`` expressions by using ``.negated`` before and after simplifying.
+    """
+    # Simplify expressions of the type Gt(x, Min/Max(..., x, ...))
+    from sympy.functions.elementary.miscellaneous import Min, Max
+    if e.has(Min, Max):
+        enew = None
+        if isinstance(e.rhs, (Min, Max)) and not e.lhs.has(Min, Max):
+            # c >/>= Min/Max, reverse
+            enew = e.reversed
+        elif isinstance(e.lhs, (Min, Max)) and not e.rhs.has(Min, Max):
+            # Min/Max >/>= c
+            enew = e
+        if enew is not None:
+            if e.rhs.is_number:
+                # Other side is number, evaluate cases
+                numberpart, remaining = sift(e.lhs.args, lambda x: x.is_number, binary=True)
+                if numberpart:
+                    if isinstance(e.lhs, Max):
+                        if cmp1(Max(*numberpart), e.rhs):
+                            enew = S.true
+                        else:
+                            enew = fn(Max(*remaining), e.rhs)
+                    else:
+                        if cmp2(Min(*numberpart), e.rhs):
+                            enew = S.false
+                        else:
+                            enew = fn(Min(*remaining), e.rhs)
+            else:
+                # Other part is Symbol/expression, evaluate cases
+                constantpart, remaining = sift(e.lhs.args, lambda x: x == e.rhs, binary=True)
+                if constantpart:
+                    if fn == StrictGreaterThan:
+                        if isinstance(e.lhs, Max):
+                            enew = StrictGreaterThan(Max(*remaining), e.rhs)
+                        else:
+                            enew = S.false
+                    else:  # GreaterThan
+                        if isinstance(e.lhs, Max):
+                            enew = S.true
+                        else:
+                            enew = GreaterThan(Min(*remaining), e.rhs)
+                            # e = GreaterThan(e.lhs._new_rawargs(*remaining), e.rhs).canonical
+        return enew
+
+def _simplify_equality_with_minmax(e):
+    """ Simplify :class:`~.Equality` that
+    contains a :class:`~.Min` and/or :class:`~.Max` functions.
+    """
+    from sympy.functions.elementary.miscellaneous import Min, Max
+    if e.has(Min, Max):
```

Comment:
The same thing can be done here -- then more targetted simplification can call this without having to worry about what type of non-inequality. Perhaps it could be renamed `_simplify_noninequality_with_minmax`?